src/coord/src/catalog/storage.rs

```diff
@@ -277,36 +275,38 @@ impl Connection {
         res
     }
 
-    pub fn get_catalog_content_version(&mut self) -> Result<usize, Error> {
+    pub fn get_catalog_content_version(&mut self) -> Result<String, Error> {
         let tx = self.inner.transaction()?;
-        let current_setting: Option<u32> = tx
+        let current_setting: Option<String> = tx
             .query_row(
-                "SELECT CAST(value AS int) FROM settings WHERE name = 'catalog_content_version';",
+                "SELECT value FROM settings WHERE name = 'catalog_content_version';",
                 params![],
                 |row| row.get(0),
             )
             .optional()?;
         let version = match current_setting {
-            Some(v) => v,
+            Some(v) => match v.parse::<u32>() {
+                // Prior to v0.8.4 catalog content versions was stored as a u32
+                Ok(_) => "pre-v0.8.4".to_string(),
+                Err(_) => v,
+            },
             None => {
                 tx.execute(
-                    "INSERT INTO settings (name, value) VALUES ('catalog_content_version', 0);",
+                    "INSERT INTO settings (name, value) VALUES ('catalog_content_version', 'new');",
                     params![],
                 )?;
-                0
+                "new".to_string()
             }
```

Comment:
Oh this is kinda weird now. There is the possibility of writing `new` into the database like this. I feel like we should ditch this branch and then use an `UPSERT` down in `set_catalog_content_version` below?